src/python/pants/backend/python/target_types.py

```diff
@@ -54,19 +54,41 @@ class PythonSources(Sources):
     expected_file_extensions = (".py", ".pyi")
 
 
-class PythonInterpreterCompatibility(StringOrStringSequenceField):
-    """A string for Python interpreter constraints on this target.
+class InterpreterConstraintsField(StringSequenceField):
+    """The Python interpreters this code is compatible with.
```

Comment:
It's a bit odd that the name and the definition swapped here... it used to be named `compatibility` and be described as "constraints on the target", and now it's named `constraints` and described as "what the code is compatible with".

Reply:
True, I think that's because both are valid ways of describing this mechanism. Now that the field's name expresses the idea of "interpreter constraints", we can switch the description to cover the point previously expressed via the field name `compatibility`.